### PR TITLE
add Suricata 7 support to Suricata

### PR DESCRIPTION
**Pull request by VVelox**
Suricata 7 is the current version of Suricata and this extend really needs updated for it.

todo..
- new test for v2 of the extend
- update docs

DO NOT DELETE THE UNDERLYING TEXT

#### Please note

> Please read this information carefully. You can run `./lnms dev:check` to check your code before submitting.

- [x] Have you followed our [code guidelines?](https://docs.librenms.org/Developing/Code-Guidelines/)
- [x] If my Pull Request does some changes/fixes/enhancements in the WebUI, I have inserted a screenshot of it.
- [x] If my Pull Request makes discovery/polling/yaml changes, I have added/updated [test data](https://docs.librenms.org/Developing/os/Test-Units/).

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

### CODE REVIEW

File: includes/html/pages/device/apps/suricata.inc.php [hunk 1]
```diff
@@ -1,45 +1,824 @@
...
+        'suricata_mem_use' => 'Memory Usage',
+        'suricata_uptime' => 'Uptime',
+        'suricata_alert' => 'Alert Status',
+    ];
+} elseif ($app_data['version'] == 2) {
+    if (! isset($vars['suricata_graph_set'])) {
+        $vars['suricata_graph_set'] = 'general';
+    }
+
+    if (strcmp($vars['suricata_graph_set'], 'general') == 0) {
```
Comment (murrant) [inline includes/html/pages/device/apps/suricata.inc.php:104]: kind of weird using strcmp() when you are just checking if they are equal

**Review by murrant — CHANGES_REQUESTED** [review]
Not anything specific but because there are so many variables, some of this is hard to read.  Try to minimize the amount of scrolling to figure out what something is.  For example, setting a bunch of variables at the top of the file only to use them once at the bottom makes it harder to read the code.

Definitely change the strcmp to == or === instead.

File: includes/html/pages/device/apps/suricata.inc.php (hunk 1 above)
Comment (VVelox) [reply includes/html/pages/device/apps/suricata.inc.php:104]: Derp! Good point. `==` is way better here. I tend to forget it works for this as well and strcmp is what comes up in the docs when I forget. :P

Comment (VVelox) [reply includes/html/pages/device/apps/suricata.inc.php]: And fixed!

**Comment by VVelox** [comment]
> Not anything specific but because there are so many variables, some of this is hard to read. Try to minimize the amount of scrolling to figure out what something is. For example, setting a bunch of variables at the top of the file only to use them once at the bottom makes it harder to read the code.

hmmm.... Extract bits of it to it's own files and the graph file contains a array that is used for dynamically building that all

**Pull request merged**